### ISSUE TITLE
Amend require section of composer.json with maximum compatibility version

### ISSUE THREAD

**Issue by mamsincl** [Smile-SA/elasticsuite#2023]
**Is your feature request related to a problem? Please describe.**
During and (especially) after Magento upgrade (via composer) the incompatibility with the previous Elasticsuite version could cause unexpected behaviour. Debugging the issue could be time-consuming and may result custom patches and unnecessary PRs

**Describe the solution you'd like**
Extend compatible version with maximum version in **require** section of composer.json

**Describe alternatives you've considered**
We can call the beginning of [README](https://github.com/Smile-SA/elasticsuite/blob/2.10.x/README.md) as an alternative solution already exists

**Additional context**
example for version 2.9.x (_compatible with Magento >=2.3.5 <2.4.0_ - linking to _magento/framework_ should be enough):
```json
"require": {
        "magento/framework": ">=102.0.5 <103.0.0",
        "magento/module-store": ">=101.0.5",
        "magento/module-backend": ">=101.0.5",
        "magento/module-catalog": ">=103.0.5",
        "magento/module-catalog-search": ">=101.0.5",
        "magento/module-catalog-graph-ql": ">=100.3.5",
        "magento/module-inventory-sales-api": ">=1.0.3",
        "magento/module-inventory-sales": ">=1.0.3",
        "magento/module-inventory-indexer": ">=1.0.3",
        "magento/magento-composer-installer": "*",
        "elasticsearch/elasticsearch": "~6.7|~7.2"
    },
```

**Comment by romainruaud** [comment]
Yep, I do agree with this, it would also save us some time answering on "invalid" issues here that are due to misfits with maximum supported versions.

**Comment by mamsincl** [comment]
I'll see if I can spare some time to help with this ... 
Can you advise on the best approach of MRs?

**Comment by romainruaud** [comment]
@mamsincl you can propose a PR to 2.9.x and 2.10.x branch

Regards
